tests/MSBuildDeviceIntegration/Tests/InstallTests.cs

```diff
@@ -504,15 +506,22 @@ public void IncrementalFastDeployment ()
 			using (var lb2 = CreateDllBuilder (Path.Combine (rootPath, lib2.ProjectName)))
 				Assert.IsTrue (lb2.Build (lib2), "Second library build should have succeeded.");
 
+			long lib1FirstBuildSize = new FileInfo (Path.Combine (rootPath, lib1.ProjectName, lib1.OutputPath, "Library1.dll")).Length;
+
 			using (var builder = CreateApkBuilder (Path.Combine (rootPath, app.ProjectName))) {
 				builder.ThrowOnBuildFailure = false;
+				builder.BuildLogFile = "install.log";
 				Assert.IsTrue (builder.Install (app), "First install should have succeeded.");
+				var logLines = builder.LastBuildOutput;
+				Assert.IsTrue (logLines.Any (l => l.Contains ("NotifySync CopyFile") && l.Contains ("UnnamedProject.dll")), "UnnamedProject.dll should have been uploaded");
+				Assert.IsTrue (logLines.Any (l => l.Contains ("NotifySync CopyFile") && l.Contains ("Library1.dll")), "Library1.dll should have been uploaded");
+				Assert.IsTrue (logLines.Any (l => l.Contains ("NotifySync CopyFile") && l.Contains ("Library2.dll")), "Library2.dll should have been uploaded");
 				var firstInstallTime = builder.LastBuildTime;
+				builder.BuildLogFile = "install2.log";
 				Assert.IsTrue (builder.Install (app, doNotCleanupOnUpdate: true, saveProject: false), "Second install should have succeeded.");
 				var secondInstallTime = builder.LastBuildTime;
 
 				var filesToTouch = new [] {
-					Path.Combine (rootPath, lib1.ProjectName, "Class1.cs"),
```

Comment:
Maybe I'm missing something, but does `Class1.cs` not ever get `Touch()` called on it?

Reply:
We change the actual TextContent here https://github.com/xamarin/xamarin-android/pull/5229/files#diff-34b4814429c211ae0bf929ddaa36bbc4eb8c13de248a744fa75e7a8f333fb9aeR533

Reply:
I think the `TimeStamp` has to change for it to save the `.cs` file?

Reply:
I'll test that out

Reply:
The timestamp is set by this line https://github.com/xamarin/xamarin-android/pull/5229/files#diff-34b4814429c211ae0bf929ddaa36bbc4eb8c13de248a744fa75e7a8f333fb9aeR539